### PR TITLE
[NO MERGE] SHM_NAME_MAX

### PR DESCRIPTION
**Pull request by adamsitnik**
I don't have an ARM64 macOS machine (yet), so I need to use a very dirty way of figuring out the magic number

Related to #63240

### CODE REVIEW

**Comment by am11** [comment]
It is a bit strange to truncate the name in retry attempt on managed side, when we know the (documented) limit in native implementation. Perhaps we need to just replace the asserts with NULL, something like (with or without condition):

```diff
--- src/native/libs/System.Native/pal_io.c
+++ src/native/libs/System.Native/pal_io.c

 #if defined(SHM_NAME_MAX) // macOS 
-    assert(strlen(name) <= SHM_NAME_MAX);
+    if (strlen(name) > SHM_NAME_MAX) name[SHM_NAME_MAX] = '\0';
 #elif defined(PATH_MAX) // other Unixes 
-    assert(strlen(name) <= PATH_MAX); 
+    if (strlen(name) > PATH_MAX) name[PATH_MAX] = '\0';
 #endif 
```

**Comment by adamsitnik** [comment]
@dotnet/area-infrastructure-libraries how can I force our CI to run the tests on arm64 macOS machines?

**Comment by adamsitnik** [comment]
> @area-infrastructure-libraries how can I force our CI to run the tests on arm64 macOS machines?

@ericstj @safern ping (I need to fix this bug as the backport is incomplete but I can't repro it using the CI or @Jozkee M1)

BTW is it expected for the staging CI leg to be green despite having test failures?

image

**Comment by ericstj** [comment]
> BTW is it expected for the staging CI leg to be green despite having test failures?

Yes, you can see that here https://github.com/dotnet/runtime/blob/4e48641996cfc890163a49ef13bb2121de58d31d/docs/pr-builds.md#runtime-staging

> how can I force our CI to run the tests on arm64 macOS machines

I think outerloop and runtime-extra-platforms contain `OSX_arm64` that but that might trigger a lot more than what you need, and I believe other resources used by those pipelines are scarce right now.  I've pinged the infrastructure channel with this request.

**Comment by ericstj** [comment]
@adamsitnik -- this should do it.

**Comment by adamsitnik** [comment]
@ericstj thank you!

btw the tests have passed: 

image

I was also not able to repro the issue using @Jozkee machine.

I'll try to re-run the CI leg and if it does not fail ask other folks for running it on their macOS arm64 hardware.

**Comment by ericstj** [comment]
@safern let me know that outerloop isn't a superset, so it won't run innerloop tests ☹️   Not sure how to explicitly trigger InnerLoop on M1.  I think you can do it manually in AzDo.
Opened https://github.com/dotnet/runtime/issues/64929 to add this to the extra-platforms pipeline.

File: src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs [hunk 1]
```diff
@@ -209,11 +210,22 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
                         // the result of native shm_open does not work well with our subsequent call to mmap.
                         return null;
                     }
+                    else if (errorInfo.Error == Interop.Error.ENAMETOOLONG)
+                    {
+                        --maxNameLength;
```
Comment (akoeplinger) [inline src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs:215]: should we try to protect against maxNameLength being 0 or negative?

**Comment by adamsitnik** [comment]
Thank you all for your help, I've opened #68541 with hopefully the proper fix.

**Pull request closed without merging**